designer/client/src/components/modals/DeployWithParametersDialog.tsx

```diff
@@ -0,0 +1,143 @@
+import { css, cx } from "@emotion/css";
+import { WindowButtonProps, WindowContentProps } from "@touk/window-manager";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+import { getProcessName, getProcessVersionId, getScenarioGraph } from "../../reducers/selectors/graph";
+import { getFeatureSettings } from "../../reducers/selectors/settings";
+import { PromptContent, WindowKind } from "../../windowManager";
+import CommentInput from "../comment/CommentInput";
+import ProcessDialogWarnings from "./ProcessDialogWarnings";
+import { FormHelperText, Typography } from "@mui/material";
+import { LoadingButtonTypes } from "../../windowManager/LoadingButton";
+import { ActionNodeParameters } from "../../types/action";
+import { AdvancedParametersSection } from "./AdvancedParametersSection";
+import { mapValues } from "lodash";
+import HttpService from "../../http/HttpService";
+import { ActionParameter } from "./ActionParameter";
+import { NodeTable } from "../graph/node-modal/NodeDetailsContent/NodeTable";
+import { ToggleProcessActionModalData } from "./DeployProcessDialog";
+
+function initialNodesData(params: ActionNodeParameters[]) {
+    return params.reduce(
+        (paramObj, { nodeId, parameters }) => ({
+            ...paramObj,
+            [nodeId]: mapValues(parameters, (value) => value.defaultValue || ""),
+        }),
+        {},
+    );
+}
+
+export function DeployWithParametersDialog(props: WindowContentProps<WindowKind, ToggleProcessActionModalData>): JSX.Element {
+    // TODO: get rid of meta
+    const {
+        meta: { action, displayWarnings },
+    } = props.data;
+    const processName = useSelector(getProcessName);
+    const processVersionId = useSelector(getProcessVersionId);
+    const scenarioGraph = useSelector(getScenarioGraph);
+    const [parametersDefinition, setParametersDefinition] = useState([]);
+    const [values, setValues] = useState({});
+
+    const getActionParameters = useCallback(async () => {
+        const response = await HttpService.getActionParameters(processName, scenarioGraph);
```

Comment:
Maybe it's worth introducing some loader indicator here, and display content only, when the initial values are correctly set. Also, what about an error scenario where FE could not get actionParameters?

Reply:
I've added loader and showBoundary when FE could not get actionParameters. ErrorBoundary is defined in the same way as in WindowContent. All of it makes that user sees spinner while parameters are loading and user sees big, full screen, error panel when an error occurs.